src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -116,20 +115,37 @@ public static async Task<ChatCompletion<T>> CompleteAsync<T>(
         ChatOptions? options = null,
         bool? useNativeJsonSchema = null,
         CancellationToken cancellationToken = default)
-        where T : class
     {
         _ = Throw.IfNull(chatClient);
         _ = Throw.IfNull(chatMessages);
         _ = Throw.IfNull(serializerOptions);
 
         serializerOptions.MakeReadOnly();
 
-        var schemaNode = AIJsonUtilities.CreateJsonSchema(
+        var schemaElement = AIJsonUtilities.CreateJsonSchema(
             type: typeof(T),
             serializerOptions: serializerOptions,
             inferenceOptions: _inferenceOptions);
 
-        var schema = JsonSerializer.Serialize(schemaNode, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonElement)));
+        var isWrappedInObject = false;
+
+        if (!SchemaRepresentsObject(schemaElement))
+        {
+            // We wrap regardless of native structured output, since it also applies to Azure Inference
+            var schemaAsJsonObject = (JsonObject)JsonSerializer.Deserialize(
+                schemaElement,
+                AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonObject)))!;
+            schemaElement = JsonSerializer.SerializeToElement(new JsonObject
```

Comment:
Given that the final result of this method is the `schema` string, couldn't we just directly serialize the wrapping `JsonObject` to that?

Reply:
Effectively
```csharp
string schema;
if (SchemaRepresentsObject(schemaElement))
{
    schema = JsonSerializer.Serialize(schemaElement, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonElement)));
}
else
{
    JsonObject wrappedSchema = ...;
    schema = JsonSerializer.Serialize(wrappedSchema, ...);
}
```

Reply:
Updated